File: poc-kronos-frontend/src/model/horario/Substituto.js
class Substituto {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.dataInicial = dados ? dados.dataInicial : null;
    this.dataFinal = dados ? dados.dataFinal : null;
    this.idFuncionario = dados ? dados.idFuncionario : null;
    this.nomeFuncionario = dados ? dados.nomeFuncionario : null;
    this.idHorario = dados ? dados.idHorario : null;
  }

  static validations(required, dataFimMenor) {
    return {
      dataInicial: { required },
      dataFinal: { required, dataFimMenor: dataFimMenor("dataInicial") },
      idFuncionario: { required },
      idHorario: { required }
    };
  }

  static validationMessages() {
    return {
      dataInicial: {
        required: "Data Início é obrigatório"
      },
      dataFinal: {
        required: "Data Final é obrigatória",
        dataFimMenor: "Data Final não pode ser menor que a Data Inicial"
      },
      idFuncionario: {
        required: "Funcionário é obrigatório"
      },
      idHorario: {
        required: "Horário é obrigatório"
      }
    };
  }
}
export default Substituto;

File: src/main/java/br/com/kronos/backend/aplicacao/horario/Substituto.java

package br.com.kronos.backend.aplicacao.horario;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Substituto {
    
    private Long id;
    private LocalDate dataInicial;
    private LocalDate dataFinal;
    private Long idFuncionario;
    private Long idHorario;

	@Builder
    public Substituto(Long id, LocalDate dataInicial, LocalDate dataFinal, Long idFuncionario, Long idHorario) {

		super();
        this.id = id;
        this.dataInicial = exigirNaoNulo(dataInicial, "Data inicial"); 
        this.dataFinal = exigirNaoNulo(dataFinal, "Data final"); 
        this.idFuncionario = exigirNaoNulo(idFuncionario, "Funcionário (professor)"); 
        this.idHorario = exigirNaoNulo(idHorario, "Horário"); 
	}
}